### DISCUSSION TITLE
groping equal positions that are sequential

### DISCUSSION THREAD

**Discussion by alessiaf12** [kernc/backtesting.py#698] [General]
Hi everyone,
I need some information. It is possible to group sequential trades with same positions (long or short) in a single trade?

Esemple
![image](https://user-images.githubusercontent.com/109291746/181266570-69dfe4e5-f9ea-447c-b0e0-d34c3bd8e609.png)
It is possible group trades 0 and 1 (both short)? Or trades 71, 72, 73 in a single trade?

I hope I explained the matter enought.

Thanks to anyone who will answer :)

**Comment by alessiaf12** [comment]
Hey @kernc I really need your help!

**Reply by kernc** [reply]
You can write your own simple grouper in Python? Writing blocks of custom Python code somewhat exceeds free support I'm willing to provide here, but the way I'd do it is use [`itertools.groupby()`](https://docs.python.org/3/library/itertools.html#itertools.groupby), which doesn't by itself sort the items, with boolean
`key=lambda x: x['Size'] > 0` (pseudo code). This should group successive entries with the same trade direction.
